internal/policies/privilege/privilege.go

```diff
@@ -279,3 +419,36 @@ func getSystemPolkitAdminIdentities(ctx context.Context, policyKitDir string) (a
 
 	return adminIdentities, nil
 }
+
+// isOldPolkit checks current polkit-1 configuration to determine if the current version < 124.
+//
+// To determine the version, we follow the steps:
+//  1. If the old configuration directory does not exist or is empty -> version < 124.
+//  2. If the old configuration directory only contains the adsys generated file -> version < 124.
+//
+// If the previous checks are valid, we still need to check if the new configuration file exists as the user
+// could have installed the compatibility package (polkitd-pkla), which adds old configuration files even if
+// the polkit version is >= 124.
+func isOldPolkit(policyKitDir, policyKitReservedDir string) bool {
+	dirEntries, err := os.ReadDir(filepath.Join(policyKitDir, "localauthority.conf.d"))
+	nEntries := len(dirEntries)
+	if err != nil || nEntries == 0 {
+		return false
+	}
+
+	if nEntries == 1 && dirEntries[0].Name() == adsysOldPolkitName+".conf" {
+		return false
+	}
+
+	if _, err := os.Stat(filepath.Join(policyKitReservedDir, "rules.d/49-ubuntu-admin.rules")); err != nil {
+		return true
+	}
+	return false
+}
+
+const (
```

Comment:
I would put those consts at the top as they are used in `ApplyPolicy`, no?

Reply:
They are, but these template-like strings are kind of "ugly" to look at, so I didn't want to throw them at someone's face for no reason 🤣